pymc_experimental/model/marginal_model.py

```diff
@@ -244,6 +255,192 @@ def marginalize(self, rvs_to_marginalize: Union[TensorVariable, Sequence[TensorV
         # Raise errors and warnings immediately
         self.clone()._marginalize(user_warnings=True)
 
+    def _to_transformed(self):
+        "Create a function from the untransformed space to the transformed space"
+        transformed_rvs = []
+        transformed_names = []
+
+        for rv in self.free_RVs:
+            transform = self.rvs_to_transforms.get(rv)
+            if transform is None:
+                transformed_rvs.append(rv)
+                transformed_names.append(rv.name)
+            else:
+                transformed_rv = transform.forward(rv, *rv.owner.inputs)
+                transformed_rvs.append(transformed_rv)
+                transformed_names.append(self.rvs_to_values[rv].name)
+
+        fn = self.compile_fn(inputs=self.free_RVs, outs=transformed_rvs)
+        return fn, transformed_names
+
+    def unmarginalize(self, rvs_to_unmarginalize):
+        for rv in rvs_to_unmarginalize:
+            self.marginalized_rvs.remove(rv)
+            self.register_rv(rv, name=rv.name)
+
+    def recover_marginals(
+        self, idata, var_names=None, return_samples=True, extend_inferencedata=True
+    ):
+        """Computes normalized posterior probabilities of marginalized variables
```

Comment:
The only reason I took the normalized was to reuse the posterior word, what is more important was to mention it also samples

```suggestion
        """Computes posterior log-probabilities and samples of marginalized variables
```